packages/dai-plugin-governance/src/utils/irv.js

```diff
@@ -0,0 +1,325 @@
+      return prv;
+    });
+
+    // mark the weakest as eliminated
+    tally.options[optionToEliminate].eliminated = true;
+
+    // a vote needs to be moved if...
+    // 1) it's currently for the eliminated candidate
+    // 2) there's another choice further down in the voter's preference list
+    const votesToBeMoved = votes
+      .map((vote, index) => ({ ...vote, index }))
+      .filter(vote => parseInt(vote.choice) === parseInt(optionToEliminate))
+      .filter(vote => vote.ballot[vote.ballot.length - 1] !== 0);
+
+    // move votes to the next choice on their preference list
+    votesToBeMoved.forEach(vote => {
+      const prevChoice = votes[vote.index].choice;
+      votes[vote.index].choice = votes[vote.index].ballot.pop();
+      if (!tally.options[votes[vote.index].choice])
+        tally.options[votes[vote.index].choice] = { ...defaultOptionObj };
+
+      if (tally.options[votes[vote.index].choice].eliminated) {
+        votes[vote.index].choice = votes[vote.index].ballot.pop();
+        let validVoteFound = false;
+
+        while (votes[vote.index].choice !== 0) {
+          if (!tally.options[votes[vote.index].choice])
+            tally.options[votes[vote.index].choice] = { ...defaultOptionObj };
+          if (!tally.options[votes[vote.index].choice].eliminated) {
+            validVoteFound = true;
+            break;
+          }
+          votes[vote.index].choice = votes[vote.index].ballot.pop();
+        }
+
+        if (!validVoteFound) return;
+      }
+      if (!tally.options[votes[vote.index].choice].eliminated) {
+        tally.options[votes[vote.index].choice].transfer = BigNumber(
+          tally.options[votes[vote.index].choice].transfer
+        ).plus(vote.mkrSupport || 0);
+
+        tally.options[prevChoice].transfer = BigNumber(
+          tally.options[prevChoice].transfer
+        ).minus(vote.mkrSupport || 0);
+      }
+    });
+
+    // look for a candidate with the majority
+    Object.entries(tally.options).forEach(
+      ([option, { firstChoice, transfer }]) => {
+        if (firstChoice.plus(transfer).gt(totalMkrParticipation.div(2)))
+          tally.winner = option;
+      }
+    );
+
+    // count the number of options that haven't been eliminated
+    const remainingOptions = Object.entries(tally.options).filter(
+      ([, optionDetails]) => !optionDetails.eliminated
+    ).length;
+
+    // if there are no more rounds,
+    // or if there is only one opiton remaining
+    // the winner is the option with the most votes
+    if (
+      (tally.rounds > MAX_ROUNDS && !tally.winner) ||
+      (remainingOptions === 1 && !tally.winner)
+    ) {
+      let max = BigNumber(0);
+      let maxOption;
+      Object.entries(tally.options).forEach(
+        ([option, { firstChoice, transfer }]) => {
+          if (firstChoice.plus(transfer).gt(max)) {
+            max = firstChoice.plus(transfer);
+            maxOption = option;
+          }
+        }
+      );
+
+      tally.winner = maxOption;
```

Comment:
in this case, shouldn't there be no winner?

Reply:
yea, that's probably correct since no options have the majority if this is reached. good call